packages/yourloops/test/integration/pages/patient/caregivers-list.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import { loggedInUserId, mockAuth0Hook } from '../../mock/mockAuth0Hook'
+import { IUser, UserRoles } from '../../../../models/user'
+import { mockNotificationAPI } from '../../mock/mockNotificationAPI'
+import { mockTeamAPI } from '../../mock/mockTeamAPI'
+import { mockUserDataFetch } from '../../mock/auth'
+import { mockPatientAPI } from '../../mock/mockPatientAPI'
+import { addDirectShareMock, mockDirectShareApi } from '../../mock/mockDirectShareAPI'
+import { act, screen, within } from '@testing-library/react'
+import { renderPage } from '../../utils/render'
+import { checkPatientLayout } from '../../assert/layout'
+import userEvent from '@testing-library/user-event'
+import DirectShareApi from '../../../../lib/share/direct-share-api'
+import { UserInvitationStatus } from '../../../../models/generic'
+import { INotification } from '../../../../lib/notifications/models'
+
+describe('Patient caregivers page', () => {
+  const firstName = 'Théo'
+  const lastName = 'Cupé'
+
+  const caregiverId = 'caregiver-id'
+  const caregiverFirstName = 'Claire'
+  const caregiverLastName = 'Hyère'
+  const caregiverEmail = 'caregiver@email.com'
+
+  beforeAll(() => {
+    mockAuth0Hook(UserRoles.patient)
+    mockNotificationAPI()
+    mockTeamAPI()
+    mockUserDataFetch(firstName, lastName)
+    mockPatientAPI()
+    mockDirectShareApi()
+  })
+
+  it('should render the caregivers page, allow to add and remove a caregiver', async () => {
+    await act(async () => {
+      renderPage('/caregivers')
+    })
+
+    checkPatientLayout(`${firstName} ${lastName}`)
+
+    const secondaryBar = screen.getByTestId('patient-caregivers-secondary-bar')
+    expect(secondaryBar).toBeInTheDocument()
+
+    const addCaregiverButton = within(secondaryBar).getByText('Add Caregiver')
+    expect(addCaregiverButton).toBeVisible()
+
+    userEvent.click(addCaregiverButton)
+
+    const addCaregiverDialog = screen.getByRole('dialog')
+    expect(addCaregiverDialog).toBeVisible()
+
+    const addCaregiverDialogTitle = within(addCaregiverDialog).getByText('Add a caregiver')
+    expect(addCaregiverDialogTitle).toBeVisible()
+
+    const addCaregiverDialogEmailInput = within(addCaregiverDialog).getByRole('textbox', { name: 'Email' })
+    expect(addCaregiverDialogEmailInput).toBeVisible()
+
+    const addCaregiverDialogConfirmButton = within(addCaregiverDialog).getByRole('button', { name: 'Invite' })
+    expect(addCaregiverDialogConfirmButton).toBeVisible()
+    expect(addCaregiverDialogConfirmButton).toBeDisabled()
+
+    const addCaregiverDialogCancelButton = within(addCaregiverDialog).getByText('Cancel')
+    expect(addCaregiverDialogCancelButton).toBeVisible()
+
+    await userEvent.type(addCaregiverDialogEmailInput, caregiverEmail)
+
+    expect(addCaregiverDialogConfirmButton).toBeEnabled()
+
+    jest.spyOn(DirectShareApi, 'getDirectShares').mockResolvedValueOnce([{
```

Comment:
opt: move spies at the beginning of the test

Reply:
As discussed, leaving it as-is